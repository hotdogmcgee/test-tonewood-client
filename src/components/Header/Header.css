.Header {
  height: 40px;
  display: flex;
  align-items: normal;
  justify-content: space-between;
}

.Header h1 {
  margin: 0;
  font-size: 16px;
  font-weight: 500
}

.Header a {
  color: #444;
  text-decoration: none !important;
}

.Header a:hover {
  color: none !important;
}

.Header__tagline--wide {
  opacity: 0.35;
  margin-left: 10px;
}

.Header__tagline--narrow {
  opacity: 0.35;
  display: block;
  text-align: center;
  display: none;
  margin-top: 10px;
}

.Header__not-logged-in {
  display: flex;
  /* flex: 1; */
  align-items: center;
  justify-content: flex-end;
}

.Header__not-logged-in a:last-child {
  background: rgb(255, 121, 121);
  color: white;
  padding: 5px 15px;
  border-radius: 15px;
  margin-left: 10px;
}

.Header__logged-in a {
  border: 1px solid rgba(0, 0, 0, 0.25);
  padding: 5px 15px;
  border-radius: 15px;
}

.Header__logged-in a:first-child {
  margin-right: 3px;
}

@media screen and (min-width: 450px) {
  .Header {
    align-items: baseline;
  }

  .Header h1 {
    font-size: 22px;
  }
}

@media screen and (max-width: 630px) {
  .Header__tagline--wide {
    display: none;
  }

  .Header__tagline--narrow {
    display: inherit;
  }
}

@media screen and (max-width: 400px) {
  .Header__logged-in a {
    padding: 5px 10px;
    font-size: 13px;
  }
}

@media screen and (min-width: 900px) {
  .Header__logged-in a:first-child {
    margin-right: 10px;
  }
  .Header__tagline--wide {
    text-align: center;
    width: 300px;
    position: absolute;
    left: 0;
    right: 0;
    margin-left: auto;
    margin-right: auto;
  }
}
