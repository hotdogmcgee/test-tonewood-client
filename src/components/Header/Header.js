import React from "react";
import { Link } from "react-router-dom";
import TokenService from "../../services/token-service";
import "./Header.css";

export default class Header extends React.Component {
  handleLogoutClick = () => {
    TokenService.clearAuthToken();
    this.props.hasLogin(false);
  };

  renderLogoutLink() {
    return (
      <div className="Header__logged-in">
        <Link onClick={this.handleLogoutClick} to="/">
          Logout
        </Link>
      </div>
    );
  }

  renderLoginLink() {
    return (
      <div className="Header__not-logged-in">
        <Link to="/login">Log in</Link>
        <Link to="/register">Register</Link>
      </div>
    );
  }

  render() {
    return (
      <>
        <nav className="Header">
          <h1>
            <Link to="/"> Tonewoods</Link>
          </h1>
          <span className="Header__tagline--wide">
            Help build the tonewood project.
          </span>
          {TokenService.hasAuthToken()
            ? this.renderLogoutLink()
            : this.renderLoginLink()}
        </nav>
        <span className="Header__tagline--narrow">
          Help build the tonewood project.
        </span>
      </>
    );
  }
}
