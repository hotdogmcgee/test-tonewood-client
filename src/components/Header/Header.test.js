import React from "react";
import Header from "./Header";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe("Header component", () => {
  it("renders a Header by default", () => {
    const wrapper = shallow(<Header />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
