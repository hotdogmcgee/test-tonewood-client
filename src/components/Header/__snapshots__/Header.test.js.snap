// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders a Header by default 1`] = `
<Fragment>
  <nav
    className="Header"
  >
    <h1>
      <Link
        to="/"
      >
         Tonewoods
      </Link>
    </h1>
    <span
      className="Header__tagline--wide"
    >
      Help build the tonewood project.
    </span>
    <div
      className="Header__not-logged-in"
    >
      <Link
        to="/login"
      >
        Log in
      </Link>
      <Link
        to="/register"
      >
        Register
      </Link>
    </div>
  </nav>
  <span
    className="Header__tagline--narrow"
  >
    Help build the tonewood project.
  </span>
</Fragment>
`;
